src/filenames/bcl_processor.rs

```diff
@@ -120,12 +119,28 @@ impl BclProcessorFileGroup {
 pub struct BclProcessorFile {
     pub group: BclProcessorFileGroup,
     pub read: String,
-    pub path: PathBuf,
+    pub full_path: PathBuf,
 }
 
-impl fmt::Display for BclProcessorFile {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{:?}", self.path)
+impl BclProcessorFile {
+    fn new(full_path: &Path) -> Option<Self> {
```

Comment:
Thanks for scooping this up into a constructor!  It would be worth adding a docstring that explains why this may return None.

Reply:
Added a docstring!